onnxruntime/python/tools/transformers/optimizer.py

```diff
@@ -174,13 +181,41 @@ def optimize_by_onnxruntime(
         else:
             providers.append("CUDAExecutionProvider")
 
-    onnxruntime.InferenceSession(onnx_model_path, sess_options, providers=providers, **kwargs)
+    if isinstance(onnx_model, str):
```

Comment:
I would add a comment here to explain why the inference session is created here for future readers.

Reply:
I don't quite understand the reason for this. Can you quickly explain this so I can add it here?

Reply:
So we realized the inference session is used for optimizing the model right? And the optimized models are just a side effect of creating this inference session. Since the session itself is unused, it would be good to explain to readers in words to clarify.